### PR TITLE
feat: ability to skip migration of certain contracts/models

### PR DESCRIPTION
**Pull request by lambda-0x**
# Description

fix: #1924

## Tests

- [x] Yes
- [ ] No, because they aren't needed
- [ ] No, because I need help

## Added to documentation?

- [ ] README.md
- [ ] [Dojo Book](https://github.com/dojoengine/book)
- [ ] No documentation needed

## Checklist

- [x] I've formatted my code (`scripts/prettier.sh`, `scripts/rust_fmt.sh`, `scripts/cairo_fmt.sh`)
- [x] I've linted my code (`scripts/clippy.sh`, `scripts/docs.sh`)
- [x] I've commented my code
- [x] I've requested a review after addressing the comments

### CODE REVIEW

File: bin/sozo/src/commands/dev.rs [hunk 1]
```diff
@@ -244,9 +248,23 @@ where
...
     }
 
-    let new_manifest =
+    let mut new_manifest =
         BaseManifest::load_from_path(&manifest_dir.join(MANIFESTS_DIR).join(BASE_DIR))?;
 
+    if let Some(skip_manifests) = skip_migration {
+        for contract_or_model in skip_manifests {
+            if let Some(index) =
+                new_manifest.contracts.iter().position(|c| c.name == contract_or_model)
```
Comment (glihm) [inline bin/sozo/src/commands/dev.rs:257]: We may use [retain](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.retain).

File: crates/dojo-bindgen/src/lib.rs [hunk 2]
```diff
@@ -111,10 +115,25 @@ fn gather_dojo_data(
...
 ) -> BindgenResult<DojoData> {
     let root_dir: Utf8PathBuf = manifest_path.parent().unwrap().into();
     let base_manifest_dir: Utf8PathBuf = root_dir.join("manifests").join(profile_name).join("base");
-    let base_manifest = BaseManifest::load_from_path(&base_manifest_dir)?;
+    let mut base_manifest = BaseManifest::load_from_path(&base_manifest_dir)?;
+
+    if let Some(skip_manifests) = skip_migration {
+        for contract_or_model in skip_manifests {
+            if let Some(index) =
+                base_manifest.contracts.iter().position(|c| c.name == contract_or_model)
```
Comment (glihm) [inline crates/dojo-bindgen/src/lib.rs:127]: Same here we could use `retain`.

File: crates/sozo/ops/src/migration/utils.rs [hunk 3]
```diff
@@ -30,6 +31,20 @@ where
     let mut local_manifest = BaseManifest::load_from_path(&profile_dir.join(BASE_DIR))
         .map_err(|e| anyhow!("Fail to load local manifest file: {e}."))?;
 
+    if let Some(skip_manifests) = skip_migration {
```
Comment (glihm) [inline crates/sozo/ops/src/migration/utils.rs:34]: Actually after reading this in lot's of places, would it be convenient to have a type to manage that? Which could have a function implemented once, and only called when required?

File: examples/spawn-and-move/src/models.cairo [hunk 4]
```diff
@@ -47,6 +47,14 @@ struct Moves {
     last_direction: Direction
 }
 
+#[derive(Copy, Drop, Serde)]
+#[dojo::model]
+struct Skipped {
```
Comment (glihm) [inline examples/spawn-and-move/src/models.cairo:52]: Could be better to have a more realistic name that is only skipped in some profile, for instance an example of a Mock that is only used with the `dev` profile?

File: examples/spawn-and-move/src/skipped.cairo [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#[dojo::contract]
+mod skipped {}
```
Comment (glihm) [inline examples/spawn-and-move/src/skipped.cairo:2]: Same here, if we could make the example more real, could be great. :)

File: crates/sozo/ops/src/migration/utils.rs (hunk 3 above)
Comment (lambda-0x) [reply crates/sozo/ops/src/migration/utils.rs:34]: good idea, added it directly in  `BaseManifest`

**Review by glihm — APPROVED** [review]
Thanks for the changes @lambda-0x! I'm just thinking that we may ignore some of the generated manifests that are not overlays and keep perhaps only the `manifest.toml` and `manifest.json` actually.

**Pull request merged**